src/methods/join.js

```diff
@@ -341,27 +352,42 @@ export class JoinImplementation {
     realm: Realm,
     joinCondition: AbstractValue,
     pnc: PossiblyNormalCompletion,
-    v: Value
+    nc: SimpleNormalCompletion
```

Comment:
Given the name of the method, do the wrapping inside it rather than at the point of call.